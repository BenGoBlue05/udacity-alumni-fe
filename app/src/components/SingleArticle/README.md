## SingleArticle Component
A component that renders a single article for the site's blog.

### Example

```js
<SingleArticle article={article} />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **article**    | Object   |             | A single article conforming to the schema defined on the API.


### Other Information
Presentational component only
