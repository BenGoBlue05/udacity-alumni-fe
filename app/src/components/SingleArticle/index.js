import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Headline from 'grommet-udacity/components/Headline';
import Section from 'grommet-udacity/components/Section';
import Box from 'grommet-udacity/components/Box';
import Article from 'grommet-udacity/components/Article';
import Image from 'grommet-udacity/components/Image';
import Markdown from 'grommet-udacity/components/Markdown';
import { Author, ArticleMeta } from 'components';

const SingleArticle = ({
  article,
}) => (
  <Box>
    <Section align="center">
      <Article align="center" className={styles.singleArticle}>
        <Headline align="center">
          {article.title}
        </Headline>
        <Image src={article.feature_image} full="true" />
        <Box
          align="center"
          justify="center"
          className="markdown-body"
          pad={{ horizontal: 'small' }}
        >
          <Markdown
            content={article.content}
            components={{
              h1: { props: { strong: true } },
              h2: { props: { strong: true } },
              p: { props: { size: 'large' } },
              img: { props: { size: 'small' } },
            }}
          />
        </Box>
      </Article>
    </Section>
    <Section align="center">
      <Author author={article.user} />
    </Section>
    <Section align="center">
      <ArticleMeta article={article} />
    </Section>
  </Box>
);

SingleArticle.propTypes = {
  article: PropTypes.object.isRequired,
};

export default cssModules(SingleArticle, styles);
