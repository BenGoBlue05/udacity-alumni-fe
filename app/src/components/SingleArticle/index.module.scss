.heading {
  padding: 20px;
}
.singleArticle {
  position: relative;
  margin: 0 auto;
  width: 90%;
  max-width: 750px;
}