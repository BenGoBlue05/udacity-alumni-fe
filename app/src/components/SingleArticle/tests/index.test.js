import SingleArticle from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<SingleArticle />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <SingleArticle />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
