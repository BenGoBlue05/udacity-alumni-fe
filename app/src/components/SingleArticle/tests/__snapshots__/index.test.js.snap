exports[`<SingleArticle /> should render with default props 1`] = `
<div>
  <Article
    align="center"
    direction="column"
    pad="none">
    <Heading
      align="center"
      tag="h2">
      Creditable coverage Hcfa common procedure coding system Group health plan
    </Heading>
    <Image
      size="medium"
      src="https://robohash.org/magnietnon.png?size=300x300" />
    <Box
      align="center"
      announce={false}
      direction="column"
      focusable={true}
      justify="center"
      pad={
        Object {
          "horizontal": "small"
        }
      }
      responsive={true}
      size="large"
      tag="div">
      <Paragraph>
        Care plan eligible dependent free-look period waiting period incurral date board certified health care provider HCPCS. Agent of record PCP limited policy morbidity medigap health savings account credentialing medical necessity open enrollment. Creditable coverage agent of record renewal disenroll IPA medicare. Accumulation period contract year policy benefit cap medicare supplement creditable coverage.
      </Paragraph>
    </Box>
  </Article>
  <undefined
    author={undefined} />
</div>
`;
